### ISSUE TITLE
The specified module could not be found.

### ISSUE THREAD

**Issue by triad-rom** [NLua/NLua#350]
Hi, 

I'm loading a 32bit C library module called `pb.dll` it was built using Lua5.1 but everytime I attach it to lua (using require) it gives me an error "The specified module could not be found."

Using a standard Lua, 
```
local pb =  package.loadlib("pb","luaopen_pb")
print ( pb )
```
I get a return `function: 6b3024a0` meaning the luaopen_pb() is there and is properly exported. 

The module is built using mingw 
```
gcc -O2 -c -o pb.o pb.c -m32
gcc -O -shared -o pb.dll pb.o -L./dir_to_lua51 -llua51 -m32
```

Any more information let me know.

Edit:

If its built with Lua53, the error showing is:
```
Unhandled Exception: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at KeraLua.NativeMethods.lua_pcallk(IntPtr luaState, Int32 nargs, Int32 nresults, Int32 errorfunc, IntPtr ctx, IntPtr k)
   at NLua.Lua.DoFile(String fileName)
   at MyProgram.Program.Main(String[] args) in D:\VS-Projects\MyProgram\MyProgram\Program.cs:line 25
```

**Comment by viniciusjarina** [comment]
NLua will use just a wrapper on top of the original Lua implementation. Try to load the module using the standalone lua.exe using Lua 5.3

**Comment by triad-rom** [comment]
Will it be possible to access the module using NLua afterwards?

**Comment by viniciusjarina** [comment]
NLua just uses the same binary of the original Lua, if you are able to load your mudule using Lua you should be able to load using NLua 👍

**Comment by triad-rom** [comment]
Solved it! It wasn't really compiled with Lua 5.3, somehow I forgot I copy-pasted the Lua5.1 to that folder. XD
